src/h3lib/lib/vertex.c

```diff
@@ -269,3 +269,28 @@ void H3_EXPORT(vertexToPoint)(H3Index vertex, GeoCoord* coord) {
     // Copy from boundary to output coord
     *coord = gb.verts[0];
 }
+
+/**
+ * Whether the input is a valid H3 vertex
+ * @param  vertex H3 index possibly describing a vertex
+ * @return        Whether the input is valid
+ */
+int H3_EXPORT(isValidVertex)(H3Index vertex) {
+    if (H3_GET_MODE(vertex) != H3_VERTEX_MODE) {
+        return 0;
+    }
+
+    int vertexNum = H3_GET_RESERVED_BITS(vertex);
+    H3Index owner = vertex;
+    H3_SET_MODE(owner, H3_HEXAGON_MODE);
+    H3_SET_RESERVED_BITS(owner, 0);
+
+    // The easiest way to ensure that the owner + vertex number is valid,
+    // and that the vertex is canonical, is to recreate and compare.
+    H3Index canonical = H3_EXPORT(cellToVertex)(owner, vertexNum);
+    if (vertex != canonical) {
+        return 0;
+    }
+
+    return H3_EXPORT(h3IsValid)(owner);
```

Comment:
May want to check that the owner is valid first?

Reply:
Hm. I guess it depends whether `h3IsValid` is faster - but it also would avoid handing invalid indexes to `cellToVertex`, so I think that makes sense.